src/Engine/Objects/Actor.cpp

```diff
@@ -4250,109 +4199,72 @@ int Actor::MakeActorAIList_BLV() {
-        if (distance < 0) distance = 0;
+        int distance = int_get_vector_length(delta_x, delta_y, delta_z) - pActors[i].uActorRadius;
+        if (distance < 0)
+            distance = 0;
 
         // actor is in range
         if (distance < 10240) {
-            pActors[i].ResetHostile();  // ~0x01000000
+            pActors[i].ResetHostile();
             if (pActors[i].ActorEnemy() || pActors[i].GetActorsRelation(0)) {
                 pActors[i].uAttributes |= ACTOR_HOSTILE;
                 if (!(pParty->GetRedAlert()) && (double)distance < 307.2)
                     pParty->SetRedAlert();
                 if (!(pParty->GetYellowAlert()) && distance < 5120)
                     pParty->SetYellowAlert();
             }
-            ai_near_actors_distances[active_actor_count] = distance;
-            ai_near_actors_ids[active_actor_count] = i;
-            active_actor_count++;
+            activeActorsDistances.push_back({i, distance});
         } else {
             // otherwise idle
             pActors[i].ResetActive();
         }
     }
 
-    // bubble sorting active actors by distance
-    if (active_actor_count > 0) {
-        for (uint i = 0; i < active_actor_count; i++) {
-            int change = 0;
-            for (uint j = 0; j < active_actor_count - i - 1; j++) {
-                if (ai_near_actors_distances[j] > ai_near_actors_distances[j + 1]) {
-                    std::swap(ai_near_actors_ids[j], ai_near_actors_ids[j + 1]);
-                    std::swap(ai_near_actors_distances[j], ai_near_actors_distances[j + 1]);
-                    change = 1;
-                }
-            }
-            // if no change the list is sorted
-            if (!change) break;
-        }
-    }
+    // sort active actors by distance
+    std::sort(activeActorsDistances.begin(), activeActorsDistances.end(), [] (std::pair<int, int> a, std::pair<int, int> b) {
+              return a.second < b.second; });
 
-    // checks nearby actors can detect player and takes nearest 30
-    int num_actors_detect_player = 0;
-    for (int i = 0; i < active_actor_count; i++) {
-        if (pActors[ai_near_actors_ids[i]].ActorNearby() ||
-            Detect_Between_Objects(
-                PID(OBJECT_Actor, ai_near_actors_ids[i]), PID(OBJECT_Player, 0))) {
-            pActors[ai_near_actors_ids[i]].uAttributes |= ACTOR_NEARBY;
-            ai_array_detected_actor_ids[num_actors_detect_player] = ai_near_actors_ids[i];
-            ai_array_detected_actor_dist[num_actors_detect_player++] = ai_near_actors_distances[i];
-            if (num_actors_detect_player >= 30) break;
-        }
-    }
+    // checks nearby actors can detect player
+    std::copy_if(activeActorsDistances.begin(), activeActorsDistances.end(), std::back_inserter(pickedActors), [&] (std::pair<int, int> val) {
+                 return pActors[val.first].ActorNearby() || Detect_Between_Objects(PID(OBJECT_Actor, val.first), PID(OBJECT_Player, 0)); });
```

Comment:
I believe the point of the code that was replaced with this line was not to call `Detect_Between_Objects` too many times. New code calls `Detect_Between_Objects` for all actors unconditionally.